server/data/data-api/src/main/java/org/apache/james/rrt/lib/Mapping.java

```diff
@@ -77,6 +78,40 @@ static IdentityMappingPolicy selectIdentityPolicy(Type type) {
         throw new IllegalStateException("unhandle enum type");
     }
 
+    enum MailAddressConversionPolicy {
```

Comment:
Maybe a functional interface is more accurate to this goal.

Reply:
Using an enum allows to : 
* limit the number of implementation
* name the functions
It better suites this case